src/java/org/apache/cassandra/service/DataResolver.java

```diff
@@ -48,8 +48,10 @@
     private static final boolean DROP_OVERSIZED_READ_REPAIR_MUTATIONS =
         Boolean.getBoolean("cassandra.drop_oversized_readrepair_mutations");
 
+    private static final UnaryOperator<PartitionIterator> IDENTITY = UnaryOperator.identity();
```

Comment:
Are you sure this is being allocated when you invoke it? The lambda defined at source should only be instantiated once, so this looks to be an unnecessary obfuscation.

Reply:
I think you're [right](https://stackoverflow.com/questions/23983832/is-method-reference-caching-a-good-idea-in-java-8). We shouldn't end up with multiple instances from the same call site below. In any case, I'll likely revert this if we proceed w/ the [guardrails approach|https://issues.apache.org/jira/browse/CASSANDRA-15907?focusedCommentId=17148207&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17148207].

Reply:
👍  - just happened to notice this when I skimmed the patch description, so thought I'd mention it